src/main/java/com/nvidia/spark/rapids/jni/kudo/OpenByteArrayOutputStream.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.jni.kudo;
+
+import ai.rapids.cudf.HostMemoryBuffer;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Arrays;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class extends {@link ByteArrayOutputStream} to provide some internal methods to save copy.
+ */
+public class OpenByteArrayOutputStream extends ByteArrayOutputStream {
+
+  /**
+   * Creates a new byte array output stream. The buffer capacity is
+   * initially 32 bytes, though its size increases if necessary.
+   */
+  public OpenByteArrayOutputStream() {
+    this(32);
+  }
+
+  /**
+   * Creates a new byte array output stream, with a buffer capacity of
+   * the specified size, in bytes.
+   *
+   * @param   size   the initial size.
+   * @exception  IllegalArgumentException if size is negative.
+   */
+  public OpenByteArrayOutputStream(int size) {
+    super(size);
+  }
+
+  /**
+   * Get underlying byte array.
+   */
+  public byte[] getBuf() {
+    return buf;
+  }
+
+  /**
+   * Get actual number of bytes that have been written to this output stream.
+   * @return Number of bytes written to this output stream. Note that this maybe smaller than length of
+   *      {@link OpenByteArrayOutputStream#getBuf()}.
+   */
+  public int getCount() {
+    return count;
+  }
+
+  /**
+   * Increases the capacity if necessary to ensure that it can hold
+   * at least the number of elements specified by the minimum
+   * capacity argument.
+   *
+   * <br/>
+   *
+   * This code is copied from jdk's implementation.
```

Comment:
should we state at the begining of the class that most of the class code is copied from jdk, and highlight what is added on ? for both copywrite and mainenance considerations